src/fu-engine.c

```diff
@@ -1829,6 +1858,102 @@ fu_engine_get_action_id_for_device (FuEngine *self,
 	return "org.freedesktop.fwupd.update-internal";
 }
 
+/**
+ * fu_engine_get_action_id_for_store:
+ * @self: A #FuEngine
+ * @device_id: A device ID
+ * @store: The #AsStore with the firmware metadata
+ * @flags: The #FwupdInstallFlags, e.g. %FWUPD_DEVICE_FLAG_UPDATABLE
+ * @error: A #GError, or %NULL
+ *
+ * Gets the PolicyKit action ID to use for the install operation.
+ *
+ * Returns: string, e.g. `org.freedesktop.fwupd.update-internal-trusted`
+ **/
+const gchar *
+fu_engine_get_action_id_for_store (FuEngine *self,
+				    const gchar *device_id,
+				    AsStore *store,
+				    FwupdInstallFlags flags,
+				    GError **error)
+{
+	FuDevice *device;
+	g_autofree gchar *guids = NULL;
+	g_autoptr(GPtrArray) devices = NULL;
+	const gchar *action = NULL, *action_old = NULL;
+	g_autoptr(GString) str = g_string_new (NULL);
+	gboolean matched = FALSE;
+
+	/* Got a specific device */
+	if (g_strcmp0 (device_id, FWUPD_DEVICE_ID_ANY) != 0) {
+		device = fu_device_list_get_by_id (self->device_list, device_id, error);
+		if (device == NULL)
+			return NULL;
+		return fu_engine_get_action_id_for_device (self, device,
+							   store, flags, error);
+	}
+
+	/* Got a wild card */
+	devices = fu_device_list_get_all (self->device_list);
+	for (guint i = 0; i < devices->len; i++) {
+		g_autoptr(GError) error_local = NULL;
+		device = g_ptr_array_index (devices, i);
+		if (fu_engine_store_get_app_by_guids (store, device) != NULL) {
+			matched = TRUE;
+			action_old = action;
+			action = fu_engine_get_action_id_for_device (self,
+								     device,
+								     store,
+								     flags,
+								     &error_local);
+			if (action == NULL) {
+				g_string_append_printf (str, "%s (%s): %s\t",
+							 fu_device_get_name (device),
+							 fu_device_get_id (device),
+							 error_local->message);
+			}
+			/* first action */
+			if (action_old == NULL && action)
+				continue;
+			/* same action */
+			if (action_old == action)
+				continue;
+			/* different actions */
+			if (action_old && action && action_old != action) {
+				g_set_error (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_INVALID_FILE,
+					     "Conflicting actions %s and %s were found",
+					     action_old, action);
+				return NULL;
+			}
+		}
+	}
+
+	if (str->len > 1)
+		g_string_truncate (str, str->len - 1);
+
+	if (action == NULL) {
+		if (matched) {
+			g_set_error (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOTHING_TO_DO,
+				     "There are no updates to install : %s",
+				     str->str);
+		} else {
+			guids = fu_engine_get_guids_from_store (store);
+			g_set_error (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INVALID_FILE,
+				     "No attached hardware matched this firmware, got: %s.",
+				     guids);
+		}
+	}
+	else
```

Comment:
braces, and I'm not exactly sure why we'd be skipping updates.

Reply:
Same as above comment in the archive might have some older components.